core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -79,18 +79,18 @@ public class TaskManagerConfig
     private Duration interruptStuckSplitTasksDetectionInterval = new Duration(2, TimeUnit.MINUTES);
 
     private boolean scaleWritersEnabled = true;
-    // Set the value of default max writer count to the number of processors and cap it to 32. We can do this
+    // Set the value of default max writer count to the number of processors * 2 and cap it to 64. We can do this
     // because preferred write partitioning is always enabled for local exchange thus partitioned inserts will never
     // use this property. Hence, there is no risk in terms of more numbers of physical writers which can cause high
     // resource utilization.
-    private int scaleWritersMaxWriterCount = min(getAvailablePhysicalProcessorCount(), 32);
+    private int scaleWritersMaxWriterCount = min(getAvailablePhysicalProcessorCount(), 32) * 2;
```

Comment:
Commit message says `same value as the number of worker threads`, but `maxWorkerThreads` is `Runtime.getRuntime().availableProcessors() * 2` which is a different number